nuxeo-services/nuxeo-platform-directory/nuxeo-platform-directory-test/src/test/java/org/nuxeo/directory/test/TestInitLoadDirectoryWithAutoIncrementId.java

```diff
@@ -0,0 +1,71 @@
+/*
+ *  (C) Copyright 2020 Nuxeo (http://nuxeo.com/) and others.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  Contributors:
+ *      Thierry Casanova
+ */
+
+package org.nuxeo.directory.test;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.core.api.DocumentModelList;
+import org.nuxeo.ecm.core.test.DefaultRepositoryInit;
+import org.nuxeo.ecm.core.test.annotations.Granularity;
+import org.nuxeo.ecm.core.test.annotations.RepositoryConfig;
+import org.nuxeo.ecm.directory.Session;
+import org.nuxeo.ecm.directory.api.DirectoryService;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.HotDeployer;
+
+/**
+ * @since 11.1
+ */
+@RunWith(FeaturesRunner.class)
+@Features({ DirectoryFeature.class })
+@RepositoryConfig(init = DefaultRepositoryInit.class, cleanup = Granularity.METHOD)
+public class TestInitLoadDirectoryWithAutoIncrementId {
+
+    protected static final String CSV_LOAD_DIRECTORY = "csvLoadedDirectory";
+
+    @Inject
+    protected DirectoryService directoryService;
+
+    @Inject
+    protected HotDeployer hotDeployer;
+
+    @Test
+    @Deploy("org.nuxeo.ecm.directory.tests:csv-on-missing-columns-and-never-load-directory-override-contrib.xml")
+    public void testInitDirectoryWithOnMissingColumnsAndAutoIncrementId() throws Exception {
+        hotDeployer.deploy(
+                "org.nuxeo.ecm.directory.tests:csv-on-missing-columns-and-update-duplicate-autoincrementId-directory-override-contrib.xml");
+        try (Session session = directoryService.open(CSV_LOAD_DIRECTORY)) {
+            // check that csv is loaded only at first time
+            Map<String, Serializable> filter = Collections.singletonMap("obsolete", false);
+            DocumentModelList entries = session.query(filter);
+            assertEquals(3, entries.size());
```

Comment:
when methode is launched thedirectory is initialized with "test-append-directory.csv" wich contains 3 entries.Then we hotrelaod with "test-load-directory.csv" wich contains 2 entries (with one duplicate". Since AutoIncrementId cannot be merge we can see that the size is unchanged (3 entries)